src/Tema1-DianaNiculescu.txt

```diff
@@ -0,0 +1,114 @@
+                                                  XPATH LOCATORS
+
+
+1) Locatorul  //button[@onclick='swapCheckbox()'] are urmatoarele:
+- tagname = button
+- attribute = onclick
+- attribute value = swapCheckbox()
+- rol: identifica in pagina "Dynamic Controls"(https://the-internet.herokuapp.com/dynamic_controls) butonul "Remove"
+
+2) Locatorul //div[@id='checkbox'] are urmatoarele:
+- tagname = div
+- attribute = id
+- attribute value = checkbox
+- rol: identifica in pagina "Dynamic Controls"(https://the-internet.herokuapp.com/dynamic_controls) checkbox-ul "A checkbox"
+
+3) Locatorul //input[@class='form-control-file'] are urmatoarele :
+- tagname = inpus
+- attribute = class
+- attribute value = form-control-file
+- rol: identifica in pagina 'Upload and Download'(https://demoqa.com/upload-download) butonul 'Choose File'
+
+4) Accesand pagina 'https://demoqa.com/text-box' am vrut sa caut elementul pentru casuta "Full Name".
+Am scris locatorul "//input[@autocomplete='off']". Acest locator imi returna doua rezultate: unul specific casutei 'Full Name', celalalt specific casutei "Email"
+Am adaugat in locator inca un atribut, pentru a-mi returna doar elementul de care aveam nevoie ,cel pentru "Full Name":  //input[@autocomplete='off'and @placeholder='Full Name']
+Contine doua atribute(autocomplete si placeholder) si tag name-ul "input"
+
+5) Locatorul //a[@id='bad-request'] are urmatoarele:
+- tagname = a
+- attribute = id
+- attribute value = bad-request
+- rol: identifica in pagina "Links"(https://demoqa.com/links) link-ul "Bad Request"
+
+6) Locatorul //button[@id='startStopButton'] are urmatoarele:
+- tagname = button
+- attribute = id
+- attribute value = startStopButton
+- rol: identifica in pagina "Widgets-Progress Bar"(https://demoqa.com/progress-bar) butonul "Start"
+
+ 7) Locatorul (//img[@src='/images/bookimage3.jpg'])[1] are urmatoarele:
```

Comment:
deobicei cand identifici un element web, urmaeza sa interactionezi cu el, sa dai click, sa iei text, sa inserezi, etc. In acest caz, ai identificat doar imaginea cartii cu care nu se poate interactiona. Ca sa interactionezi cu aceasta carte(web element) e nevoie sa mergi la link sau buton (orice e intaractionabil cu mouse-ul).